src/client/dfs/dfs.c

```diff
@@ -562,14 +562,14 @@ insert_entry(daos_handle_t oh, daos_handle_t th, const char *name, size_t len,
 	sgl.sg_iovs	= sg_iovs;
 
 	rc = daos_obj_update(oh, th, flags, &dkey, 1, &iod, &sgl, NULL);
-	if (rc) {
-		/** don't log error if conditional failed */
-		if (rc != -DER_EXIST && rc != -DER_NO_PERM)
+
+	if ((rc != 0) && (rc != -DER_EXIST)) {
```

Comment:
yea we should not log non-daos logic errors in DFS. can print them as debug.
there is a problem here though.. we are returning DER_EXIST, but it should be errno instead since other calls are expecting errno. so this seems wrong to me.
also i don't get the reason for this change here.. can you please explain?

Reply:
I agree. EEXIST is returned in the updated code. thx. I need to change here, otherwise D_ERROR() prints error due to existing entry. Now EEXIST will handled in fs_copy_symlink().